File: javascript/leetcode/single-number-iii.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var singleNumber = function(nums) {
    seen = {}
    for (var i = 0; i < nums.length; i++) {
        if (seen[nums[i]]) {
            seen[nums[i]]++
        } else {
            seen[nums[i]] = 1
        }
    }
    result = []
    for (k in seen) {
        if (seen[k] == 1) {
            result.push(parseInt(k))
        }
    }
    return result
};

File: ruby/leetcode/single-number-iii.rb
# @param {Integer[]} nums
# @return {Integer[]}
def single_number(nums)
    seen = {}
    nums.each do |i|
        seen[i] = if seen[i] then seen[i] + 1 else 1 end
    end
    result = []
    seen.each do |k, v|
        if v == 1
            result << k
        end
    end
    return result
end